sel/src/Sel/Hashing/Password.hs

```diff
@@ -95,13 +95,13 @@ instance Display PasswordHash where
 instance Eq PasswordHash where
   (PasswordHash ph1) == (PasswordHash ph2) =
     unsafeDupablePerformIO $
-      foreignPtrEq ph1 ph2 cryptoPWHashStrBytes
+      foreignPtrEq (Foreign.castForeignPtr ph1) (Foreign.castForeignPtr ph2) cryptoPWHashStrBytes
```

Comment:
since casting functions like `fromIntegral`, `castForeignPtr` `castPt` are the Devil, I'd like it if you could insert type applications in order to make sure that there is no inference happening and that they are converting to and from the types we expect.